### ISSUE TITLE
jc --asciitable changes column capitalization

### ISSUE THREAD

**Issue by shuffman** [kellyjonbrazil/jc#255]
```
$ echo "
> abcDef
> ------
> a" | jc --asciitable
[{"abcdef":"a"}]
```

This is unexpected and means when I've tried to insert `jc` into existing data pipelines, I run into issues later in the pipeline.

**Comment by kellyjonbrazil** [comment]
Hi there!

This is working as designed since the column names are normalized (convert to lowercase and spaces and special characters are converted to underscores). What is the issue you are running into downstream?

**Comment by shuffman** [comment]
Thanks Kelly -- The issue we've had is we have a data-flow that assumes certain column names. So for example we have existing legacy tools that query a batch compute system and output a table of results. Further downstream tools make assumptions about column names based on the column names produced by the original tool (for example, `ExecHost`).

Recently I've been trying to add some data filtering to this flow with jc's `--asciitable` and `jq`, then recreating the original table with `jtbl`, but then the parsing tools that made assumptions about the original column names from the batch system are failing. I can fix this by adding `sed` into the flow to replace the column names, but I would have expected `jc` to retain the original column names.

Is there something in the json spec that requires this normalizing? How would `jc` handle an (admittedly contrived) case where a table contains two columns whose names differ only in capitalization? E.g.:


``` echo "A  a
> -  -
> 5  4" | jc --asciitable
[{"a":"⁣-"},{"a":"⁣4"}```

**Comment by kellyjonbrazil** [comment]
Gotcha - there are a couple options here. One, we can take the `asciitable` parser, modify it, and make it a custom plugin for this usecase. I think it's just modifying a couple lines of code to do this.

I'm also looking at potentially moving some of the key normalization (lowercase and converting special chars) to the `_process` function. This means that by default the parser will give you the current behavior, but by using `jc --asciitable -r` you will get the non-normalized keynames.

The quick and dirty solution looks like it would be to modify the `_snake_case` function so it looks like this:
```
def _snake_case(line: str) -> str:
    """
    Replace spaces between words and special characters with an underscore
    and set to lowercase. Ignore the replacement char (�) used for header
    padding.
    """
    # line = re.sub(r'[^a-zA-Z0-9� ]', '_', line)  # special characters
    line = re.sub(r'\b \b', '_', line)           # spaces between words
    # line = line.lower()
    return line
```
(basically commenting a couple lines)

Then put the updated file in the [plugin folder](https://github.com/kellyjonbrazil/jc#custom-parsers).

**Comment by kellyjonbrazil** [comment]
This is now available in v1.20.1 available on PIP. (binaries and packages will be available soon)

**Comment by shuffman** [comment]
Awesome thanks! This will be a big help.
